pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -803,6 +808,11 @@ public CompletableFuture<LookupDataResult> newLookup(ByteBuf request, long reque
                     waitingLookupRequests.size())));
             }
         }
+        future.whenComplete((lookupDataResult, throwable) -> {
+            if (throwable instanceof ConnectException) {
```

Comment:
Why only the `ConnectException` should be handled?
IMO, should we release the semaphore for all exceptions except the `TooManyRequestsException`?

Reply:
@codelipenghui The other case will be released in `getAndRemovePendingLookupRequest`. We only need to handle https://github.com/apache/pulsar/blob/252ededebe59d7974f821b8b87fc05859ecc6f22/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java#L293